### PR TITLE
feat(replay): Add Mask and Unmask Android implementation

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] New feature

## :scroll: Description

This PR adds new and legacy architecture impl of Mask and Unmask wrappers for custom redact of React Native components.

- This is a follow up to https://github.com/getsentry/sentry-react-native/pull/4224

- This PR needs sentry-android https://github.com/getsentry/sentry-java/pull/3881

- [Legacy Arch Sample Replay](https://sentry-sdks.sentry.io/replays/437e2f8a9c024f7faffccfaa46f31ae2)
- [New Arch Sample Replay](https://sentry-sdks.sentry.io/replays/f1deb3af06ff4e8481c4812e62d2e8ac)

| Emulator | Replay |
|--------|--------|
| android-custom-mask-emulator | android-custom-mask-replay | 

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

**Comment by krystofwoldrich** [comment]
@antonis @lucas-zimerman This is a follow up to https://github.com/getsentry/sentry-react-native/pull/4224, but this time for Android.

**Review by lucas-zimerman — APPROVED** [review]
Thank you for the PR! LGTM!

File: samples/react-native/src/App.tsx [hunk 1]
```diff
@@ -277,6 +277,11 @@ function App() {
     <>
       <RootNavigationContainer />
       <RunningIndicator />
+      <Text>
+        <Text>
+          Hello
```
Comment (antonis) [inline samples/react-native/src/App.tsx:282]: q: Do we want the `Hello` text at the bottom in the sample or is it just for testing?

<details><summary>Screenshot</summary>

Screenshot_20241125_155719
</details>

Comment (krystofwoldrich) [reply samples/react-native/src/App.tsx:282]: God catch, thank you, now I'll remove it.

File: packages/core/android/src/newarch/java/io/sentry/react/replay/RNSentryReplayUnmaskManager.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    return REACT_CLASS;
+  }
+
+  @NonNull
+  @Override
+  public RNSentryReplayUnmask createViewInstance(@NonNull ThemedReactContext context) {
+    return new RNSentryReplayUnmask(context);
+  }
+
+  public static final String REACT_CLASS = "RNSentryReplayUnmask";
```
Comment (antonis) [inline packages/core/android/src/newarch/java/io/sentry/react/replay/RNSentryReplayUnmaskManager.java:35]: q: Could we reuse `RNSentryReplayUnmaskManagerImpl.REACT_CLASS` here?

**Review by antonis — APPROVED** [review]
LGTM 🚀 

Added a questions/nitpick with https://github.com/getsentry/sentry-react-native/pull/4265/files#r1856782186

File: packages/core/android/src/newarch/java/io/sentry/react/replay/RNSentryReplayUnmaskManager.java (hunk 2 above)
Comment (krystofwoldrich) [reply packages/core/android/src/newarch/java/io/sentry/react/replay/RNSentryReplayUnmaskManager.java]: Yes, definitely, thank you, I've missed this place, for mask it's already done.

**Pull request merged**